include/qunit.hpp

```diff
@@ -61,17 +86,114 @@ struct QEngineShard {
         , isEmulated(false)
         , isProbDirty(true)
         , isPhaseDirty(true)
-        , amp0(complex(ONE_R1, ZERO_R1))
-        , amp1(complex(ZERO_R1, ZERO_R1))
+        , amp0(ONE_CMPLX)
+        , amp1(ZERO_CMPLX)
         , isPlusMinus(false)
+        , controlsShards()
+        , targetOfShards()
     {
     }
 
     ~QEngineShard()
     {
-        if (unit)
+        if (unit && (mapped == 0)) {
             unit->Finish();
+        }
+    }
+
+    void RemovePhaseControl(QEngineShardPtr p)
```

Comment:
Probably need some commentary throughout all of this talking about phase usage, how it's used, what it's used for, etc.

Reply:
It has no public API, but I will make a comment to indicate that's phase gate "gate fusion," which is just a convenient internal cost-savings for QUnit.

Reply:
Done.